### PR TITLE
Added some tools and updated the Old!

### PR DESCRIPTION
**Pull request by janaSunrise**
I've edited `tools.py`, `main.py`, `customs.py` and `constants.py`.
- `main.py` and `constants.py` have minimal changes. 
- `tools.py` and `customs.py` have few more imports and several functions were added, one functions edited ig.

### CODE REVIEW

File: cogs/custom.py [hunk 1]
```diff
@@ -13,7 +15,7 @@ def __init__(self, client):
     # Commands
     @commands.command()
     async def hello(self, ctx):
-        await ctx.send('Hey there Buddy!')
+        await ctx.send(f'Hey there Buddy! My Creator is {constants.creator}')
```
Comment (ItsDrike) [inline cogs/custom.py:18]: It seems irrelevant to mention the creator name when greeting.

File: cogs/custom.py [hunk 2]
```diff
@@ -25,6 +27,25 @@ async def ping(self, ctx):
         duration = (end - start) * 1000
         await message.edit(embed=info(f":ping_pong: {duration:.2f}ms", ctx.me, "Pong!"))
 
+    @commands.command(aliases=("poll",))
+    async def vote(self, ctx: commands.Context, title: str, *options: str) -> None:
+        """
+        Build a quick voting poll with matching reactions with the provided options.
+        A maximum of 20 options can be provided, as Discord supports a max of 20
+        reactions on a single message.
```
Comment (ItsDrike) [inline cogs/custom.py:35]:
```suggestion
        Build a quick voting poll with matching reactions with the provided options.
        
        A maximum of 20 options can be provided, as Discord supports a max of 20
        reactions on a single message.
```
Multiline docstrings should consist of a "title" line separated from the rest of the docstring which is considered as "description"

File: cogs/tools.py [hunk 3]
```diff
@@ -134,6 +138,62 @@ async def ask_question(self, ctx, conversation_mode="true", *, question: str):
 
         await ctx.send(embed=embed)
 
+    @commands.command()
+    async def charinfo(self, ctx, *, characters: str):
+        """Shows you information about a number of characters. Only up to 25 characters at a time."""
```
Comment (ItsDrike) [inline cogs/tools.py:143]:
```suggestion
        """Shows you information about up to 25 characters."""
```
A bit more compact docstring

File: cogs/tools.py [hunk 4]
```diff
@@ -134,6 +138,62 @@ async def ask_question(self, ctx, conversation_mode="true", *, question: str):
...
+            name = unicodedata.name(c, 'Name not found.')
+            return f'`\\U{digit:>08}`: {name} - {c} \N{EM DASH} <http://www.fileformat.info/info/unicode/char/{digit}>'
+
+        msg = '\n'.join(map(to_string, characters))
+        if len(msg) > 2000:
+            return await ctx.send('Output too long to display.')
+        await ctx.send(msg)
+
+    @commands.command()
+    async def charinfo_uni(self, ctx: commands.Context, *, characters: str) -> None:
```
Comment (ItsDrike) [inline cogs/tools.py:155]: Why is this the only type-hinted function? Either don't type hint at all, or type-hint everywhere.
Full type hinting will be added in PR #11 until that is merged it's not necessary.
Same goes for line 176

Comment (janaSunrise) [reply cogs/tools.py:155]: i will remove the type hint, u edit it later.

File: cogs/custom.py (hunk 1 above)
Comment (Theodore-Robinson) [reply cogs/custom.py]: It needs improvement by a **lot**.
Well, it doesn't even need to be there if you ask me since a dummy command is of no use for the development team nor is it of use for people using the bot. We need to get some meaningful commands in.

Comment (janaSunrise) [reply cogs/custom.py]: @Theodore-Robinson I Changed the code just now. Review That!

File: cogs/tools.py (hunk 4 above)
Comment (Theodore-Robinson) [reply cogs/tools.py]: Yeah, we need to get type-hints and we also need to fill up metadata for the bot.
Also, we'll need to add docstrings for each function, class etc for better usability.

Comment (janaSunrise) [reply cogs/tools.py]: @Theodore-Robinson we're Editing and adding the Type Hints and i am Trying to implement the docstrings everywhere

File: cogs/tools.py [hunk 5]
```diff
@@ -134,6 +138,62 @@ async def ask_question(self, ctx, conversation_mode="true", *, question: str):
...
+            await ctx.send(embed=embed)
+            return
+
+        if len(characters) > 25:
+            embed = discord.Embed(title=f"Too many characters ({len(characters)}/25)")
+            embed.colour = discord.Colour.red()
+            await ctx.send(embed=embed)
+            return
+
+        def get_info(char: str) -> Tuple[str, str]:
```
Comment (ItsDrike) [inline cogs/tools.py:176]: You've removed the type hints from 155, but not from here, they need to be removed from here too.

Comment (janaSunrise) [reply cogs/tools.py:176]: Please send a suggestion, to do it instantly

Comment (ItsDrike) [reply cogs/tools.py:176]: @janaSunrise  Just change it manually, I'm not sure what that `m` is referring to, I assume it's member, but it's on multiple lines so there is no need to clutter the PR with more suggestions, manual commit should do it for this.

File: cogs/tools.py [hunk 6]
```diff
@@ -47,21 +46,6 @@ async def ask_question(self, ctx: Context, conversation_mode: str = "true", *, q
...
-            return f"`\\U{digit:>08}`: {name} - {c} \N{EM DASH} <http://www.fileformat.info/info/unicode/char/{digit}>"
-
-        msg = "\n".join(map(to_string, characters))
-        if len(msg) > 2000:
-            await ctx.send("Output too long to display.")
-            return
-        await ctx.send(msg)
-
     @command()
     async def charinfo_uni(self, ctx: Context, *, characters: str) -> None:
```
Comment (ItsDrike) [inline cogs/tools.py:50]: Since `charinfo` is removed, this should be renamed to `charinfo` not `charinfo_uni`, if you need `charinfo_uni` you can make an alias to the command

File: cogs/custom.py [hunk 7]
```diff
@@ -2,17 +2,18 @@
...
 from discord import Color, Embed, Member
-from discord.ext.commands import Bot, Cog, Context, command, has_permissions
+from discord.ext.commands import BadArgument, Bot, Cog, Context, command, has_permissions
 
 
 class Custom(Cog):
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
 
+    # Commands
```
Comment (ItsDrike) [inline cogs/custom.py:12]:
```suggestion
```
This shouldn't be here. That comment is useless as it's pretty obvious that these are commands.

File: cogs/tools.py [hunk 8]
```diff
@@ -6,6 +6,10 @@
 from utils.mathscrape import get_math_results
 from utils.wolframscrape import get_wolfram_data
 
+import unicodedata
+from typing import Tuple
+import re
+
```
Comment (ItsDrike) [inline cogs/tools.py:12]: These imports need to be sorted.

File: cogs/utils/constants.py [hunk 9]
```diff
@@ -2,6 +2,7 @@
 
 from discord import Color
 
+creator = "TheOriginalDude#0585"
```
Comment (ItsDrike) [inline cogs/utils/constants.py:5]:
```suggestion
creator = "The-Codin-Hole team"
```
It seems a bit unfair to take all the credit, we are all working on it now.
